pelican/__init__.py

```diff
@@ -319,6 +320,12 @@ def parse_arguments():
                         help='Relaunch pelican each time a modification occurs'
                         ' on the content files.')
 
+    parser.add_argument('--print-settings', dest='print_settings',
+                        action='append', nargs='?', metavar='SETTING_NAME',
+                        help='Print current configuration settings and exit. '
+                        'Append a setting name as argument to see the value '
+                        'for that specific setting only.')
+
```

Comment:
Isn't `nargs='*'` with default `action='store'` more appropriate?

Logic afterwards will change though.
```shell
pelican # no --print-settings
# -> args.print_settings == None

pelican --print-settings
# -> args.print_settings == []

pelican --print-settings FOO BAR
# -> args.print_settings == ['FOO', 'BAR']
```

And if you want to get fancy and avoid putting logic inside `main()`, you can subclass [`Action`](https://docs.python.org/3/library/argparse.html#argparse.Action) then do stuff in `__call__` and at the end call `parser.exit()` to terminate. In that case `action=MyActionSubclass`

Reply:
If I'm not mistaken, with the current setup you have to do
```shell
pelican --print-settings FOO --print-settings BAR # ...
```
in order to specify multiple options.

Reply:
Thanks for the input. I hadn't planned on supporting multiple setting arguments, but I added a commit that should now support invocations such as:

    pelican --print-settings MARKDOWN ARTICLE_SAVE_AS

I agree that putting logic inside `main()` is less than ideal, but when I tried subclassing Action and putting the logic inside `__call__` instead, it wasn't clear to me how to access the various variables that are plainly available in the `main()` context. It started to feel like a slippery rabbit hole, so I aborted. If you can provide more detail as to how the logic would actually fit into `__call__`, I'm certainly willing to try again.

Reply:
That did indeed point me in the right direction. Thanks! I pushed another commit that moves the logic out of `main()`. Any other thoughts on this before it's merged?

Reply:
Looks good :)